### PR TITLE
Login vs dashboard

### PR DESCRIPTION
**Pull request by cecoppinger**
Set up vue-router and used the beforeCreate lifecycle hook to direct users to the correct route based on whether they are logged in or not.

### CODE REVIEW

**Comment by ryanis-online** [comment]
I think vue-router was already added? Could you merge master into your branch and rebase to see if that fixes the merge conflict?

**Comment by cecoppinger** [comment]
Okay I think I fixed all the merge conflicts.  It's sort of working right now.  I can't see the different dashboards but the conditional routing works.

File: frontend/src/App.vue [hunk 1]
```diff
@@ -7,10 +7,25 @@
...
+    Dashboard
+  },
+  data() {
+    return {
+      user: null
+    }
+  },
+  beforeCreate: function () {
+    if (this.$store.state.currentUser.isLoggedIn === true)
+      this.$router.push('/dasboard')
```
Comment (iktani) [inline frontend/src/App.vue]:
```suggestion
      this.$router.push('/dashboard')
```

**Comment by iktani** [comment]
I know we made some changes to the main store file to use modules, but this works for me if I add a global state (with the currentUser object) back to the main store.js file. I suggested a fix to App.vue to correct a typo in the routing. Both of those changes gets me to route it properly (based on isLoggedIn) to either show a dashboard or to the placeholder for login. From a user experience standpoint, I'm not sure if the routing should immediately happen as soon as you land on the home page, but that can be changed later on.

**Comment by zstephenj** [comment]
I get the quick fixes just to make sure things are working. Regarding the modules, if you create a new User module in the store, its state can be exposed via 

`this.$store.state.User.currentUser.isLoggedIn === true)`

As it is right now, we would need the global state.

**Review by iktani — APPROVED** [review]
Looks good. Approved!

**Review by ryanis-online — APPROVED** [review]
Nice job!

**Review by simpson1986 — APPROVED** [review]

**Comment by ryanis-online** [comment]
Looks like we've got a merge conflict. I'll go ahead and merge this whenever that gets resolved

**Review by zstephenj — APPROVED** [review]

**Pull request merged**